src/index.ts

```diff
@@ -126,90 +260,73 @@ const plugin: JupyterFrontEndPlugin<void> = {
-              loadingIndicator.style.position = 'fixed';
-              loadingIndicator.style.bottom = '20px';
-              loadingIndicator.style.right = '20px';
-              loadingIndicator.style.padding = '10px';
-              loadingIndicator.style.backgroundColor = '#f0f0f0';
-              loadingIndicator.style.borderRadius = '5px';
-              loadingIndicator.style.zIndex = '1000';
-              document.body.appendChild(loadingIndicator);
-
-              await sharingService.authenticate();
-
-              let shareResponse;
-              if (isNewShare) {
-                shareResponse = await sharingService.share(notebookContent, password);
-              } else if (notebookId) {
-                shareResponse = await sharingService.update(notebookId, notebookContent, password);
-              }
+            if (result.button.accept) {
+              const { notebookName } = result.value as IShareDialogData;
+              const password = generatePassword();
 
-              if (shareResponse && shareResponse.notebook) {
-                // We need to update the metadata in the notebookContent first
-                // to do this, and we need to ensure that the metadata object exists
-                if (!notebookContent.metadata) {
-                  notebookContent.metadata = {};
-                }
+              try {
+                await sharingService.token;
+                const shareResponse = await sharingService.share(notebookContent, password);
+
+                if (shareResponse && shareResponse.notebook) {
+                  if (!notebookContent.metadata) {
+                    notebookContent.metadata = {};
+                  }
 
-                notebookContent.metadata.sharedId = shareResponse.notebook.id;
-                notebookContent.metadata.readableId = shareResponse.notebook.readable_id;
-                notebookContent.metadata.sharedName = notebookName;
-                notebookContent.metadata.isPasswordProtected = true;
+                  notebookContent.metadata.sharedId = shareResponse.notebook.id;
+                  notebookContent.metadata.readableId = shareResponse.notebook.readable_id;
+                  notebookContent.metadata.sharedName = notebookName;
+                  notebookContent.metadata.isPasswordProtected = true;
 
-                notebookPanel.context.model.fromJSON(notebookContent);
-              }
+                  notebookPanel.context.model.fromJSON(notebookContent);
+                  await notebookPanel.context.save();
+                }
 
-              let shareableLink = '';
-              if (shareResponse && shareResponse.notebook) {
-                const id = shareResponse.notebook.readable_id || shareResponse.notebook.id;
-                shareableLink = sharingService.makeRetrieveURL(id).toString();
-              }
+                let shareableLink = '';
+                if (shareResponse && shareResponse.notebook) {
```

Comment:
This is already being checked for just above - can we just combine these if/else blocks?